sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -47,6 +48,7 @@ abstract class FileFormatDataWriter(
   protected val MAX_FILE_COUNTER: Int = 1000 * 1000
   protected val updatedPartitions: mutable.Set[String] = mutable.Set[String]()
   protected var currentWriter: OutputWriter = _
```

Comment:
It seems all `OutputWriter` implementations have a path string. Shall we simply add a `def path: String` in `OutputWriter`? Then we don't need the `currentPath`

Reply:
@cloud-fan - makes sense. Af first place I was hesitating to make broader change of interface `OutputWriter`. But updated now.